backend/src/categories/categories.controller.ts

```diff
@@ -1,25 +1,142 @@
-import { Controller, Get, Body, Post, Req, Param } from '@nestjs/common';
+import {
+  Controller,
+  Get,
+  Body,
+  Post,
+  Req,
+  Param,
+  NotFoundException,
+  Put,
+  Delete,
+} from '@nestjs/common';
 import { CategoriesService } from './categories.service';
 import { CreateCategoryDto } from './dto/category-create.dto';
-
+import { AddWordDto } from './dto/add-word.dto';
+import { WordsService } from './words.service';
+import { ChoicesService } from './choices.service';
+import { HttpException, HttpStatus } from '@nestjs/common';
+import { In } from 'typeorm';
 @Controller()
 export class CategoriesController {
-  constructor(private categoriesService: CategoriesService) {}
-
+  constructor(
+    private categoriesService: CategoriesService,
+    private wordsService: WordsService,
+    private choicesService: ChoicesService,
+  ) {}
+  // Reminder: Add authguard
   @Get(['student/categories', 'admin/categories'])
   async all() {
     return this.categoriesService.find({});
   }
-
-  @Get(['student/categories/words', 'admin/categories/words'])
-  async allWords() {
-    return this.categoriesService.find({
+  // Reminder: Add authguard
+  @Get(['student/:categoryId/words', 'admin/:categoryId/words'])
+  async allWords(@Param('categoryId') categoryId: number) {
+    return await this.categoriesService.findOne({
+      where: {
+        category_id: categoryId,
+      },
       relations: ['words'],
     });
   }
-
+  // Reminder: Add authguard
   @Post('admin/category/create')
   async create(@Body() body: CreateCategoryDto) {
     return this.categoriesService.save(body);
   }
+
+  // Reminder: Add authguard
+  @Get([
+    'student/:categoryId/words/:wordsId',
+    'admin/:categoryId/words/:wordsId',
+  ])
+  async getChoices(
+    @Param('categoryId') categoryId: number,
+    @Param('wordsId') wordsId: number,
+  ) {
+    const words = await this.wordsService.findOne({
+      where: { word_id: wordsId, category_id: categoryId },
```

Comment:
Same idea as my previous comment, please remove all instances of categoryId

Just remove `categoryId`